Dockerfile

```diff
@@ -1,6 +1,6 @@
 # Inspired by https://github.com/simdjson/simdjson/blob/master/Dockerfile
 
-FROM ubuntu:23.10
+FROM ubuntu:23.04
```

Comment:
This didn't build today using 23.10, which is apparently a dev release (in the prior PR I just used the largest Ubuntu I saw on Docker Hub).  Looks like 23.04 is the `rolling` stable release, while the earlier 22.04 is the `latest` LTS based on the [Ubuntu image docs](https://hub.docker.com/_/ubuntu?uuid=D2F7152D-A7D7-47DD-BE46-1136E3260C79).